packages/core/src/state-management/repository.ts

```diff
@@ -1,70 +1,110 @@
-import { DEFAULT_LOAD_DOCOPTS, Document } from '../document';
 import DocID from '@ceramicnetwork/docid';
-import { AsyncLruMap } from './async-lru-map';
-import { DocumentFactory } from './document-factory';
-import { DocOpts, DocState, DocStateHolder } from '@ceramicnetwork/common';
+import { AnchorService, AnchorStatus, Context, DocOpts, DocState, DocStateHolder } from '@ceramicnetwork/common';
 import { PinStore } from '../store/pin-store';
-import { NetworkLoad } from './network-load';
 import { NamedTaskQueue } from './named-task-queue';
+import { DiagnosticsLogger } from '@ceramicnetwork/logger';
+import { ExecutionQueue } from './execution-queue';
+import { RunningState } from './running-state';
+import { LRUMap } from 'lru_map';
+import { Subject } from 'rxjs';
+import { StateManager } from './state-manager';
+import type { Dispatcher } from '../dispatcher';
+import type { ConflictResolution } from '../conflict-resolution';
+import type { HandlersMap } from '../handlers-map';
+import type { StateValidation } from './state-validation';
+
+export type RepositoryDependencies = {
+  dispatcher: Dispatcher,
+  pinStore: PinStore,
+  context: Context,
+  handlers: HandlersMap
+  anchorService: AnchorService,
+  conflictResolution: ConflictResolution,
+  stateValidation: StateValidation
+}
 
 export class Repository {
-  readonly loadingQ: NamedTaskQueue = new NamedTaskQueue();
-  readonly #map: AsyncLruMap<Document>;
-  #documentFactory?: DocumentFactory;
+  readonly loadingQ: NamedTaskQueue;
```

Comment:
I think it'd be helpful if all of these members of Repository had brief comments on what they represent

Reply:
Ok.